server/cache/resource_cache.go

```diff
@@ -0,0 +1,62 @@
+package cache
+
+import (
+	"context"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/listers/core/v1"
+)
+
+type ResourceCache struct {
+	ctx    context.Context
+	cache  Cache
+	client kubernetes.Interface
+	v1.ServiceAccountLister
+}
+
+func NewResourceCacheWithTimeout(client kubernetes.Interface, ctx context.Context, namespace string, timeout time.Duration) *ResourceCache {
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(client, time.Minute*20, informers.WithNamespace(namespace))
+	cache := &ResourceCache{
+		ctx:                  ctx,
+		cache:                NewLruTtlCache(timeout, 2000),
+		client:               client,
+		ServiceAccountLister: informerFactory.Core().V1().ServiceAccounts().Lister(),
+	}
+	informerFactory.Start(ctx.Done())
```

Comment:
I don't think `Start` or `Run` funcs should ever appear within a `New` func. In Java, starting "threads" in a constructor is bad-practice.

Reply:
should I add a `init` method in `resource_cache` and call if after `New` ?

Reply:
Yes. Or Run.

Reply:
done